components/webcompat_reporter/browser/webcompat_reporter_service.cc

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) 2024 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/webcompat_reporter/browser/webcompat_reporter_service.h"
+
+#include <memory>
+#include <optional>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/values.h"
+#include "brave/components/version_info/version_info.h"
+#include "brave/components/webcompat_reporter/browser/webcompat_report_uploader.h"
+
+namespace {
+constexpr char kComponentItemName[] = "name";
+constexpr char kComponentItemId[] = "id";
+constexpr char kComponentItemVersion[] = "version";
+
+void ConvertCompsToValue(
+    std::optional<base::Value>& val_to_set,
+    const std::vector<webcompat_reporter::mojom::ComponentInfoPtr>&
+        components) {
```

Comment:
```
void ConvertCompsToValue(
    const std::vector<webcompat_reporter::mojom::ComponentInfoPtr>&
        components,
    std::optional<base::Value>& val_to_set)
```
output variables should go after input ones.

why not just return an optional value?